app/lib/ui/components/snackbars/snackbars.dart

```diff
@@ -0,0 +1,168 @@
+          body: _Body()),
+    );
+  }
+}
+
+class _Body extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Padding(
+            padding: EdgeInsets.all(spacingM),
+            child: Text(
+              'Customize the snackbar before displaying it',
+              style: Theme.of(context).textTheme.bodyLarge,
+            ),
+          ),
+          _SnackBarsVariants(),
+        ],
+      ),
+    );
+  }
+}
+
+class _SnackBarsVariants extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final ComponentSnackbarsCustomizationState? customizationState =
+        ComponentSnackbarsCustomization.of(context);
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        SizedBox(
+          width: double.infinity,
+          child: TextButton(
+            onPressed: () {
+              if (customizationState?.hasTwoLines == false &&
+                  customizationState?.hasLongerAction == false) {
+                OdsSnackbars.showSnackbarSingleLine(
+                  context: context,
+                  content: AppLocalizations.of(context)!
+                      .componentSnackbarsSingleText,
+                  label: customizationState?.hasActionButton == true
+                      ? 'Close'
+                      : null,
+                  onPressed: customizationState?.hasActionButton == true
+                      ? () {}
+                      : null,
+                );
+              } else if (customizationState?.hasTwoLines == true &&
+                  customizationState?.hasLongerAction == false) {
+                OdsSnackbars.showSnackbarTwoLines(
+                  context: context,
+                  content: AppLocalizations.of(context)!
+                      .componentSnackbarsTwoLineActionText,
+                  label: customizationState?.hasActionButton == true
+                      ? 'Close'
+                      : null,
+                  onPressed: customizationState?.hasActionButton == true
+                      ? () {}
+                      : null,
+                );
+              }
+              if (customizationState?.hasLongerAction == true) {
+                OdsSnackbars.showSnackbarLongerAction(
+                  context: context,
+                  content: AppLocalizations.of(context)!
+                      .componentSnackbarsTwoLineLongerActionText,
+                  label: 'Longer action',
+                  onPressed: () {},
+                );
+              }
+            },
+            child: Text(
+              'Show snackbar',
+              style: Theme.of(context).textTheme.headlineSmall,
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class _CustomizationContent extends StatelessWidget {
```

Comment:
I made a test:
The text is on tow lines, but two  line switch is off : 
![image](https://github.com/Orange-OpenSource/ods-flutter/assets/15976893/8a8f0643-eb4a-486c-8014-6e5184b8ad64)

Reply:
on merge puis on voit avec les designer. A corriger par la suite avec un ticket de bug si besoin